x-pack/plugins/ml/public/application/components/anomalies_table/links_menu.js

```diff
@@ -53,6 +53,51 @@ class LinksMenuUI extends Component {
     };
   }
 
+  openInDiscover = async () => {
+    const {
+      services: { share },
+    } = this.props.kibana;
+    const discoverLocator = share.url.locators.get('DISCOVER_APP_LOCATOR');
+    const job = mlJobService.getJob(this.props.anomaly.jobId);
+
+    const index = job.datafeed_config.indices[0];
+    const dataViewId = (await getDataViewIdFromName(index)) || index;
+
+    const record = this.props.anomaly.source;
+    const from = timeFormatter(record.timestamp); // e.g. 2016-02-08T16:00:00.000Z
+    const to = timeFormatter(record.timestamp + record.bucket_span * 1000);
```

Comment:
I think it would be good to use the same logic defined in `openCustomUrl` in this file as we have for user-defined custom URLs with 'auto' time range setting for determining the `from` and `to` for the Discover link. This uses the aggregation interval set in the table:

- day - show full day in Discover
- hour - shows an hour either side of the hour in which the anomaly is in
- show all - shows just the selected bucket

Reply:
Updated here https://github.com/elastic/kibana/pull/120450/commits/d6e4b9adae89b65416ff57ac59a43136b49e508b

Reply:
There seems to be an issue when the Kibana timezone `dateFormat:tz` advanced setting is set to something other than the default (browser), where the time range opened in Discover doesn't correctly adjust for the timezone in the same way that the existing custom URL functionality does. I think the adjustment is added twice e.g. viewing by hour in table, timezone set to +8, anomaly at Feb 10 00:00, Discover opens to show Feb 10 07:00-09:59, where it should be Feb 9 23:00 - Feb 10 01:59.

Reply:
I was able to reproduce this as well. This hopefully should be fixed with https://github.com/elastic/kibana/pull/120450/commits/d03dfd4aa2ff76d933a7ee09c3fd416dd3245dde.